### DISCUSSION TITLE
Datahike Console

### DISCUSSION THREAD

**Discussion by den1k** [replikativ/datahike#543] [Ideas]
I'm currently throwing together a makeshift console for a project ([GRID](https://theshopgrid.com/)) and realized that I'll need this time and time again on every major project using Datahike.

This issue is meant to start the conversation around a larger effort of creating a console-type GUI for Datahike.

Datomic has a [console](https://docs.datomic.com/on-prem/console.html), Postgres has [pgAdmin](https://www.pgadmin.org/) and there are many more.

Datomic's console is written in JavaFX which is very convenient because it can run queries directly on the JVM. With Datahike being CLJC, that approach would only cater to its JVM user base. On the other hand, a javascript targeted console would work perfectly for CLJS but requires networking serialization of queries, query-results and special treatment for lazy objects like `d/entity` for CLJ.

The way I see it, there are several paths:
1. Separate console implementations for CLJ and CLJS on respective target platforms
2. An initial CLJ-only console using JavaFX or cljfx under the assumption that Clojure/JVM is the more data-intensive and durable use case
3. A joint web-based implementation in CLJC

I'm happy to help with the design and implementation as much as time allows but would also love to hear the maintainers' and community's ideas on how to approach building a console for Datahike.

**Comment by kordano** [comment]
Happy to hear that. I started working last month on a simple JavaFx prototype that interacts with datahike based on some examples by @vlaaad in his `cljfx` project. On the other hand we started with a remote server with a REST interface and were sketching a frontend in cljs with fulcro for that. Let's have maybe a call on that to discuss further.

**Comment by den1k** [comment]
Sounds good. How about Friday or Monday? I'm in NYC. If you're in Heidelberg you're 6h ahead. 5pm your time ok?

**Comment by kordano** [comment]
Monday at 5pm looks good. Just ping me on slack.

**Comment by whilo** [comment]
@den1k Just throwing this in here, in case you are interested we are now also working on https://github.com/homebaseio/datalog-console. We want this to be able to work with different Datomic Datalog databases.

**Comment by den1k** [comment]
Yes thanks I’ve been following it!
